### PR TITLE
Dotnet hexagonal

### PR DESCRIPTION
**Pull request by dhelper**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dotnet-test-samples/hexagonal-architecture/template.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Architectures: [arm64]
+    Runtime: dotnet6
+    Timeout: 30
+    Tracing: Active
+
+Resources:
+  StocksConverterFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ./src/GetStock/bin/Release/net6.0
```
Comment (jeastham1993) [inline dotnet-test-samples/hexagonal-architecture/template.yaml:16]: What compiles the application code into this folder? Typically, I have seen the CodeUri pointing to the folder containing csproj and SAM runs the build

File: dotnet-test-samples/hexagonal-architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+It uses the [IClassFixture](https://xunit.net/docs/shared-context) feature of [xUnit](https://xunit.net/) to perform setup and teardown logic. The setup code creates an DynamoDB client with the endpoint url.
+
+## Cleanup
+
+Run the given command to delete the resources that were created. It might take some time for the CloudFormation stack to get deleted.
+```
+sam delete
+```
+
+## Requirements
```
Comment (jeastham1993) [inline dotnet-test-samples/hexagonal-architecture/README.md:204]: Move this section up to the top, next to where the deployment happens

File: dotnet-test-samples/hexagonal-architecture/serverless.template [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (jeastham1993) [inline dotnet-test-samples/hexagonal-architecture/serverless.template:1]: What is the purpose of this file as well as the SAM template?

**Review by jeastham1993 — CHANGES_REQUESTED** [review]

File: dotnet-test-samples/hexagonal-architecture/README.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+### Sign up for 3rd party service
+We need a 3rd party service to retrieve real-time currencies value for this example, you can use a service like [fixer.io](https://fixer.io/), Create a free account and get the API Key used for consume the API
+
+* In the solution folder update [serverless.template](./serverless.template) file, replace ```<INSERT API KEY>``` with the fixer.io API Key.
+* In the integration test folder update [HttpClientFixtrue.cs](./tests/GetStock.IntegrationTests/Fixtures/HttpClientFixture.cs) - replace ```<INSERT API KEY>``` with the fixture.io API key.
+
+### Deploy to AWS
+The SAM template contains all the information to deploy AWS resources (an API Gateway, Lambda function and a DynamoDB table) and also the permission required by these services to communicate.
+
+You will be able to create and delete the CloudFormation stack using the SAM CLI.
```
[the remark this answers is not in the document]
Comment (dhelper) [reply dotnet-test-samples/hexagonal-architecture/README.md:50]: Will do,
But Should we explain how to use SAM every single time?

File: dotnet-test-samples/hexagonal-architecture/README.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+### Integration Tests 
+The goal of these tests is to test work along with external dependencies. In a hexagonal architecture these dependencies are called *adapters*. This sample has two such dependencies we need to test:
+* [External http based API](./tests/GetStock.IntegrationTest/Adapters/HttpClientTests.cs)
+* [DynamoDB table](./tests/GetStock.IntegrationTest/Adapters/StockDynamoDbTests.cs)
+
+Test fixtures has been implemented to setup and teardown these dependencies and remove boilerplate code fomr the tests.
+* [HttpClientFixture.cs](./tests/GetStock.IntegrationTest/Fixtures/HttpClientFixture.cs) create a client to send and receive information from 3rd party API and dispose it after test run.
+* [DynamoDbTestBase.cs](./tests/GetStock.IntegrationTest/Fixtures/DynamoDbTestBase.cs) create a client to read and store data in dynamoDB, the dynamoDB integration tests can run using [DynamoDB Local](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html) or using a real dynamoDb table on AWS. Choosing between the two options is done by setting ```RUN_LOCAL_DB``` on [using.cs](./tests/GetStock.IntegrationTest/Usings.cs)
```
[the remark this answers is not in the document]
Comment (dhelper) [reply dotnet-test-samples/hexagonal-architecture/README.md:181]: Maybe, this is the first sample we publish

File: dotnet-test-samples/hexagonal-architecture/template.yaml (hunk 1 above)
Comment (dhelper) [reply dotnet-test-samples/hexagonal-architecture/template.yaml:16]: You're right, fixed

**Review by jeastham1993 — APPROVED** [review]

**Pull request merged**